src/hoc/withAuthRedirect.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+const mapStateToPropsForRedirect = (state) => ({
+  isAuth: state.auth.isAuth,
+});
+
+const withAuthRedirect = (Component) => {
+  class RedirectComponent extends React.Component {
```

Comment:
Is it possible to have a functional component here?

Reply:
It is not possible yet

Reply:
Why? Could you explain the reason why you have a class instead of a function here?